lib/retriable.rb

```diff
@@ -6,14 +6,22 @@
 module Retriable
   module_function
 
-  def self.configure
+  def configure
     yield(config)
   end
 
   def config
     @config ||= Config.new
   end
 
+  def retriable_with_context(context_key, options = {}, &block)
```

Comment:
yeah you'd have to deprecate it and then wait until the next major version... even then it might not be worth breaking people's shit.  

if you were considering that approach though, you could deprecate it now.